src/components/ui/date-field.tsx

```diff
@@ -0,0 +1,149 @@
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+
+interface DateFieldProps {
+  date?: Date;
+  onChange?: (date?: Date) => void;
+  disabled?: (date: Date) => boolean;
```

Comment:
let's not do callback for disabled, this will lock out the users from entering once a out of range date is entered.